src/BenchmarksApps/TechEmpower/PlatformBenchmarks/BenchmarkApplication.Fortunes.cs

```diff
@@ -1,125 +1,74 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO.Pipelines;
-using System.Text.Encodings.Web;
+using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
+using RazorSlices;
 
 namespace PlatformBenchmarks
 {
     public partial class BenchmarkApplication
     {
-        private readonly static AsciiString _fortunesPreamble =
-            _http11OK +
-            _headerServer + _crlf +
-            _headerContentTypeHtml + _crlf +
-            _headerContentLength;
-
         private async Task FortunesRaw(PipeWriter pipeWriter)
         {
-            OutputFortunes(pipeWriter, await RawDb.LoadFortunesRows());
+            await OutputFortunes(pipeWriter, await RawDb.LoadFortunesRows(), FortunesTemplateFactory);
         }
 
         private async Task FortunesDapper(PipeWriter pipeWriter)
         {
-            OutputFortunes(pipeWriter, await DapperDb.LoadFortunesRows());
+            await OutputFortunes(pipeWriter, await DapperDb.LoadFortunesRows(), FortunesDapperTemplateFactory);
         }
 
         private async Task FortunesEf(PipeWriter pipeWriter)
         {
-            OutputFortunes(pipeWriter, await EfDb.LoadFortunesRows());
+            await OutputFortunes(pipeWriter, await EfDb.LoadFortunesRows(), FortunesEfTemplateFactory);
         }
 
-        private void OutputFortunes(PipeWriter pipeWriter, List<Fortune> model)
+        private ValueTask OutputFortunes<TModel>(PipeWriter pipeWriter, TModel model, SliceFactory<TModel> templateFactory)
```

Comment:
Might as well just return Task here and then pass through in the other methods if you are just going to await in those ones?

Reply:
Those methods also need to await the LoadFortunesRows() calls so you can't just return the Task.

Reply:
nvm you have to await anyway